tests/test_remote_cache.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+import os
+import shutil
+
+import pytest
+from fiona.errors import DriverError
+
+from intake_geopandas import GeoJSONSource, ShapefileSource
+
+
+def try_clean_cache(item):
+    c = None
+    for c in ['blockcache', 'filecache', 'simplecache']:
+        if c in item.storage_options:
+            caching = c
+    assert c is not None, 'caching not found'
+    path = item.storage_options[caching]['cache_storage']
+    if isinstance(path, str):
+        if os.path.exists(path):
+            shutil.rmtree(path)
```

Comment:
this caching function is similar to `clean_cache`, but doesnt reload cache https://filesystem-spec.readthedocs.io/en/latest/_modules/fsspec/implementations/cached.html

When running the same test twice as in the end of https://gist.github.com/aaronspring/3a865b991f40fa7b3afef102df0717f8, `fsspec.open_local` still tries to find the local files although deleted before

@martindurant is there a way how to call `clear_cache` in the tests or checking explicitly in `open_local` whether files are present?

Reply:
The cache implementations I thought would check for file existence and download again is not present. Sounds like a good test case that should be in the fsspec test suite.

Reply:
raised https://github.com/intake/filesystem_spec/issues/361